platform/error/platform-error-page-injector/src/main/java/org/codice/ddf/platform/error/injector/ErrorPageInjector.java

```diff
@@ -0,0 +1,210 @@
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.servlet.ServletContext;
+import org.codice.ddf.platform.error.servlet.ErrorServlet;
+import org.eclipse.jetty.servlet.ErrorPageErrorHandler;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.servlet.ServletHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.Constants;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceEvent;
+import org.osgi.framework.ServiceReference;
+import org.osgi.framework.hooks.service.EventListenerHook;
+import org.osgi.framework.hooks.service.ListenerHook.ListenerInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ErrorPageInjector implements EventListenerHook {
+
+  private final String errorPagePath = "/ErrorServlet";
+  private Map<String, String> errorCodesMap;
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ErrorPageInjector.class);
+
+  public ErrorPageInjector() {
+    errorCodesMap =
+        Stream.of("400", "401", "403", "404", "405", "406", "500", "501", "502", "503", "504")
+            .collect(
+                Collectors.collectingAndThen(
+                    Collectors.toMap(Function.identity(), x -> errorPagePath),
+                    Collections::unmodifiableMap));
+  }
+
+  Optional<BundleContext> getContext() {
+    final Bundle cxfBundle = FrameworkUtil.getBundle(ErrorPageInjector.class);
+    return (cxfBundle != null) ? Optional.of(cxfBundle.getBundleContext()) : Optional.empty();
+  }
+
+  @Override
+  public void event(ServiceEvent event, Map<BundleContext, Collection<ListenerInfo>> listeners) {
+    if (event.getType() == ServiceEvent.REGISTERED) {
+      Bundle refBundle = event.getServiceReference().getBundle();
+      BundleContext bundlectx = refBundle.getBundleContext();
+      Object service = bundlectx.getService(event.getServiceReference());
+      if (service instanceof ServletContext) {
+        injectErrorPage((ServletContext) service, refBundle);
+      }
+    }
+  }
+
+  public void init() {
+    checkForMissedServletContexts();
+  }
+
+  private void checkForMissedServletContexts() {
+
+    Optional<BundleContext> optionalBundleContext = getContext();
+    if (!optionalBundleContext.isPresent()) {
+      LOGGER.warn(
+          "Problem retrieving the bundle `Platform :: Error :: Page Injector` after it's just been initialized");
+      return; // This should never happen
+    }
+
+    //  Get all registered servletContexts in OSGI
+    final BundleContext context = optionalBundleContext.get();
+    getServiceReferences(context)
+        .ifPresent(e -> (Arrays.asList(e)).forEach(this::addMissedErrorPage));
```

Comment:
✏️ This is nitpicky, but I think the parentheses around  `Arrays.asList(e)` are unnecessary.